packages/search/src/test/use-fuzzy-search.tsx

```diff
@@ -0,0 +1,63 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import Search, { useFuzzySearch } from '..';
+import type { UseFuzzySearchOptions } from '../use-fuzzy-search';
+
+const TestComponent = < T, >( props: UseFuzzySearchOptions< T > ) => {
+	const { results, setQuery } = useFuzzySearch( props );
+
+	return (
+		<>
+			<Search onSearch={ setQuery } />
+			{ results.map( ( result ) => JSON.stringify( result ) ) }
+		</>
+	);
+};
+
+describe( 'useFuzzySearch', () => {
+	it( 'allows fuzzy searching a simple string array', () => {
+		const data = [ 'ok', 'hello' ];
+
+		render( <TestComponent data={ data } /> );
+
+		const searchbox = screen.getByRole( 'searchbox' );
+		userEvent.type( searchbox, 'k' );
```

Comment:
Ideally, in each test, we should first use `userEvent.setup()` first and then use the event triggering helpers provided by it. See p4TIVU-ac2-p2 for more details - scroll down to the "User actions" section.

This comment applies to several instances in this file.